runtime/j9vm/jvm.c

```diff
@@ -3991,10 +3998,25 @@ JVM_LoadLibrary(const char *libName, jboolean throwOnFailure)
 		JavaVM *vm = (JavaVM *)javaVM;
 		(*vm)->GetEnv(vm, (void **)&env, JNI_VERSION_1_2);
 		if (NULL != env) {
-			char errMsg[512];
-			jio_snprintf(errMsg, sizeof(errMsg), "Failed to load library \"%s\"", libName);
-			errMsg[sizeof(errMsg) - 1] = '\0';
+			char errBuf[512];
+			char *errMsg = errBuf;
+			int bufSize = sizeof(errBuf);
+			const char *portMsg = attemptedLoad ? j9error_last_error_message() : "";
+			bufSize = jio_snprintf(errMsg, bufSize, "Failed to load library (\"%s\") %s", libName, portMsg);
```

Comment:
Perhaps a minor nit, but the resulting message will end with a space if `portMsg` is empty; maybe the we should keep the format alternatives?

Reply:
Perhaps like this?
```c
			const char *errFormat = ('\0' != *portMsg)
					? "Failed to load library (\"%s\") %s"
					: "Failed to load library (\"%s\")";
			bufSize = jio_snprintf(errMsg, bufSize, errFormat, libName, portMsg);
```

Reply:
It seems quite minor for something that would likely only occur in an OOM condition, it didn't seem worth the effort. But if you insist then I will fix it.

Reply:
Yes, perhaps unlikely to occur, but the delta is simple enough I think we should fix it.

Reply:
Pushed